File: backend/services/core/src/main/java/app/mnema/core/user/config/UserClientProps.java
package app.mnema.core.user.config;

import jakarta.validation.constraints.NotBlank;
import org.springframework.boot.context.properties.ConfigurationProperties;
import org.springframework.validation.annotation.Validated;

@Validated
@ConfigurationProperties(prefix = "app.user")
public record UserClientProps(
        String baseUrl,
        @NotBlank String internalToken
) {
}

File: backend/services/auth/src/main/kotlin/app/mnema/auth/config/UserClientProps.kt
package app.mnema.auth.config

import jakarta.validation.constraints.NotBlank
import org.springframework.boot.context.properties.ConfigurationProperties
import org.springframework.validation.annotation.Validated

@Validated
@ConfigurationProperties(prefix = "app.user")
data class UserClientProps(
    val baseUrl: String = "http://localhost:8084/api/user",
    @field:NotBlank val internalToken: String = ""
)
